src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -340,6 +340,11 @@ public DiagnosticQueue DiagnosticQueue
             }
         }
 
+        /// <summary>
+        /// Callback to add additional files to be generated to disk.  Null if generating files is not desired.
+        /// </summary>
+        public Action<(string filePath, SourceText text)>? AddOutputFile;
```

Comment:
Reason this is taking a tuple versus just being an `Action<string, text>`?

Reply:
(if it's to name the parameters I both sympathize and so wish we had a language feature for this...)

Reply:
it was for naming.  i found the pair of string/SourceText not at all clear :-/